src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
The universal ref signals to me that sometimes appendObject returns a value - not a reference. When it does like, a similar block would look like the following (I understand how the universal ref differs from this code - I'm emphasizing that Object is a value in the block):
```
{
    Object obj(appendObject(txns)); // Note Object is not a ref
    obj[jss::tx_blob] = strHex (item->peekData ());
}
```
I understand the code works, and I'm not asking for a change. But I still think code in the block above is a strange idiom - take a value (not a ref), change it, and leave the block. If I replace the assignment with a serializing op like `<<` it no longer looks strange to me.